projects/plugins/jetpack/json-endpoints/class.wpcom-json-api-site-settings-endpoint.php

```diff
@@ -270,6 +270,52 @@ protected function get_cast_option_value_or_null( $option_name, $cast_callable )
 		return call_user_func( $cast_callable, $option_value );
 	}
 
+	/**
+	 * Used for the WooCommerce address options. This function returns the option value if it exists, else it looks
+	 * for a default address on the domain contact information. If that does not exist it returns an empty string.
+	 *
+	 * @param string $key The option key to retrieve a value.
+	 *
+	 * @return string The requested address part value or an empty string.
+	 */
+	protected function get_woocommerce_address( $key ) {
+		$address_part = get_option( $key );
+
+		// If there's any value set in the database, return it. Even if it's null or an empty string.
+		// False means no value for this option was found in the database.
+		if ( false !== $address_part ) {
+			return $address_part;
+		}
+
+		// If no value is set in the database, return the corresponding address part from the user's domain contact
+		// address. If that does not exist return an empty string.
+		$address_part = '';
+		if ( defined( 'IS_WPCOM' ) && IS_WPCOM ) {
+			$address = Domain_Contact_Information_Mapper::find_by_user_id( get_current_user_id() );
+
+			if ( $address ) {
+				switch ( $key ) {
+					case 'woocommerce_store_address':
+						$address_part = $address->get_address_1();
+						break;
+					case 'woocommerce_store_address_2':
+						$address_part = $address->get_address_2();
+						break;
+					case 'woocommerce_store_city':
+						$address_part = $address->get_city();
+						break;
+					case 'woocommerce_default_country':
+						$address_part = $address->get_country_code();
```

Comment:
Is this just the country code, or country-state/province?

Reply:
It's just country code. It returns 'US' for me, and $address->get_state_name() returns 'Texas' for me. But I need to do some more testing. I'm not sure how consistent this data is. Maybe it returns 'TX' as the state for someone else, for example? I'm not sure.

I'm not sure how easily we'll be able to map this to our dropdown. For example, 'US:Texas' won't match a value in our dropdown.